### PR TITLE
Always send AOD attributes from Daily Docket page

### LINKED ISSUE #14130: Editing AOD from DailyDocket sends partial update
## Description

There is a bug that is potentially creating bad AOD data in the database on the Daily Docket page. I think this stems from the fact that we are only sending changed data to the API server. I'm seeing partial AOD data that looks like this:

```
#<AdvanceOnDocketMotion:0x000055d28e40e4b0
  id: 4,
  created_at: Tue, 28 Apr 2020 18:22:16 UTC +00:00,
  granted: nil,
  person_id: 450,
  reason: "financial_distress",
  updated_at: Tue, 28 Apr 2020 18:22:16 UTC +00:00,
  user_id: nil>
```

Validations should help prevent bad data, but we should fix the underlying cause

## Acceptance criteria

- [ ] Ensure that a fully populated AdvanceOnDocketMotion record gets created

## Background/context/resources

Additional context: https://github.com/department-of-veterans-affairs/caseflow/pull/14102#discussion_r416835612

https://dsva.slack.com/archives/C3EAF3Q15/p1588101793486800

## Technical notes

This is related to how we are doing a diff before sending data to the server. This was a change we had implemented to support virtual hearings.

### PR DESCRIPTION
**Pull request by ferristseng**
Resolves #14130 

### Description

This is a very specific scenario where the AOD doesn't change if it's being switched from "Age" to a different reason. I don't know if this comes up at all in practice, but it was leading to the granted field being `nil`, and I think we are seeing bad AOD motions in prod with a similar issue.

---

**Before:**

Screen Recording 2020-05-05 at 5 06 30 PM mov

---

**After:**

Screen Recording 2020-05-05 at 5 02 41 PM mov

### CODE REVIEW

File: app/models/hearing.rb [hunk 1]
```diff
@@ -125,7 +125,10 @@ def aod?
 
   def advance_on_docket_motion
     # we're only really interested if the AOD was granted
-    AdvanceOnDocketMotion.for_person(claimant_id).order("granted DESC NULLS LAST").first
+    AdvanceOnDocketMotion
+      .for_person(claimant_id)
+      .order("granted DESC NULLS LAST, created_at DESC")
```
Comment (ferristseng) [inline app/models/hearing.rb:130]: Consider `created_at` when returning the AOD. The last AOD that was granted is relevant (note, this means you can't remove AODs).

File: app/models/hearings/forms/hearing_update_form.rb [hunk 2]
```diff
@@ -20,7 +20,10 @@ def hearing_updated?
   private
 
   def update_advance_on_docket_motion
-    AdvanceOnDocketMotion.create_or_update_by_appeal(hearing.appeal, advance_on_docket_motion_attributes)
+    AdvanceOnDocketMotion.create_or_update_by_appeal(
+      hearing.appeal,
+      advance_on_docket_motion_attributes.merge(user: RequestStore[:current_user])
```
Comment (ferristseng) [inline app/models/hearings/forms/hearing_update_form.rb:25]: Actually populate the user. Was seeing nil `user_id` previously ([see comment](https://github.com/department-of-veterans-affairs/caseflow/pull/14102#discussion_r416835612))

File: client/app/hearings/components/dailyDocket/DailyDocketRow.jsx [hunk 3]
```diff
@@ -181,25 +181,28 @@ class DailyDocketRow extends React.Component {
       return;
     }
 
-    const hearing = deepDiff(this.state.initialState, this.props.hearing);
+    const hearingChanges = deepDiff(this.state.initialState, this.props.hearing);
```
Comment (ferristseng) [inline client/app/hearings/components/dailyDocket/DailyDocketRow.jsx:184]: Changes in this file are formatting + naming. Originally I had made changes here, then opted to handle everything in `DailyDocketContainer` instead

File: app/models/hearings/forms/hearing_update_form.rb (hunk 2 above)
Comment (sahalliburton) [reply app/models/hearings/forms/hearing_update_form.rb:25]: Great catch!

**Pull request merged**